robot/framesystem/framesystem.go

```diff
@@ -66,85 +71,86 @@ func (svc *frameSystemService) Name() resource.Name {
 	return internalFrameSystemServiceName
 }
 
+// Config is a slice of *config.FrameSystemPart.
+type Config struct {
+	resource.TriviallyValidateConfig
+	Parts                []*referenceframe.FrameSystemPart
+	AdditionalTransforms []*referenceframe.LinkInFrame
+}
+
+// String prints out a table of each frame in the system, with columns of name, parent, translation and orientation.
+func (cfg Config) String() string {
+	t := table.NewWriter()
+	t.AppendHeader(table.Row{"#", "Name", "Parent", "Translation", "Orientation", "Geometry"})
+	t.AppendRow([]interface{}{"0", referenceframe.World, "", "", "", ""})
+	for i, part := range cfg.Parts {
+		pose := part.FrameConfig.Pose()
+		tra := pose.Point()
+		ori := pose.Orientation().EulerAngles()
+		geomString := ""
+		if gc := part.FrameConfig.Geometry(); gc != nil {
+			geomString = gc.String()
+		}
+		t.AppendRow([]interface{}{
+			fmt.Sprintf("%d", i+1),
+			part.FrameConfig.Name(),
+			part.FrameConfig.Parent(),
+			fmt.Sprintf("X:%.0f, Y:%.0f, Z:%.0f", tra.X, tra.Y, tra.Z),
+			fmt.Sprintf(
+				"Roll:%.2f, Pitch:%.2f, Yaw:%.2f",
+				utils.RadToDeg(ori.Roll),
+				utils.RadToDeg(ori.Pitch),
+				utils.RadToDeg(ori.Yaw),
+			),
+			geomString,
+		})
+	}
+	return t.Render()
+}
+
 // the frame system service collects all the relevant parts that make up the frame system from the robot
 // configs, and the remote robot configs.
 type frameSystemService struct {
 	resource.Named
 	resource.TriviallyCloseable
-	r      robot.Robot
-	logger golog.Logger
+	components map[string]resource.Resource
+	logger     golog.Logger
 
-	partsMu     sync.RWMutex
-	localParts  framesystemparts.Parts                     // gotten from the local robot's config.Config
-	offsetParts map[string]*referenceframe.FrameSystemPart // gotten from local robot's config.Remote
+	parts   []*referenceframe.FrameSystemPart
+	partsMu sync.RWMutex
 }
 
 // Reconfigure will rebuild the frame system from the newly updated robot.
 // NOTE(RDK-258): If remotes can trigger a local robot to reconfigure, you can cache the remoteParts in svc as well.
-// NOTE(erd): this doesn't utilize the dependencies and instead uses the robot which we would rather avoid.
-func (svc *frameSystemService) Reconfigure(ctx context.Context, _ resource.Dependencies, _ resource.Config) error {
+func (svc *frameSystemService) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
 	svc.partsMu.Lock()
 	defer svc.partsMu.Unlock()
 
-	ctx, span := trace.StartSpan(ctx, "services::framesystem::Update")
-	defer span.End()
```

Comment:
I see we removed a bunch of these traces, do we no longer use this?

Reply:
Ok I added them back